playbook/app/pb_kits/playbook/pb_drawer/_drawer.tsx

```diff
@@ -120,39 +129,54 @@ const Drawer = (props: DrawerProps): React.ReactElement => {
       }
     };
 
-    window.addEventListener('resize', handleResize);
+    window.addEventListener("resize", handleResize);
 
     // Call handler once on mount to set initial state
     handleResize();
 
     return () => {
-      window.removeEventListener('resize', handleResize);
+      window.removeEventListener("resize", handleResize);
     };
   }, [breakpoint]);
 
-  const modalIsOpened = trigger ? triggerOpened : (opened || isBreakpointOpen);
+  const modalIsOpened = trigger ? triggerOpened : opened || isBreakpointOpen;
+
+  const [animationState, setAnimationState] = useState("");
+
+  useEffect(() => {
+    if (modalIsOpened) {
+      setAnimationState("afterOpen");
+    } else if (!modalIsOpened && animationState === "afterOpen") {
+      setAnimationState("beforeClose");
+      setTimeout(() => {
+        setAnimationState("");
+      }, 200); // closeTimeoutMS
+    }
+  }, [modalIsOpened]);
+
+  const isModalVisible = modalIsOpened || animationState === "beforeClose";
 
   useEffect(() => {
-    const sizeMap: Record<DrawerProps['size'], string> = {
-      xl: '365px',
-      lg: '300px',
-      md: '250px',
-      sm: '200px',
-      xs: '64px',
+    const sizeMap: Record<DrawerProps["size"], string> = {
+      xl: "365px",
+      lg: "300px",
+      md: "250px",
+      sm: "200px",
+      xs: "64px",
     };
-    const body = document.querySelector('body');
+    const body = document.querySelector("body");
 
-    if (modalIsOpened && behavior === 'push' && body) {
-      if (placement === 'left') {
+    if (modalIsOpened && behavior === "push" && body) {
+      if (placement === "left") {
         body.style.cssText = `margin-left: ${sizeMap[size]} !important; margin-right: '' !important;`;
-      } else if (placement === 'right') {
+      } else if (placement === "right") {
         body.style.cssText = `margin-right: ${sizeMap[size]} !important; margin-left: '' !important;`;
       }
 
-      body.classList.add('ReactModal__Body--open');
+      body.classList.add("ReactModal__Body--open");
```

Comment:
Is there a reason, we are still tapping into the classes that get shipped from the ReactModal library? We shouldn't be needing these anymore.

Reply:
i changed the name to PBDrawer__Body--open